barman/backup.py

```diff
@@ -697,6 +701,23 @@ def backup(self, wait=False, wait_timeout=None, name=None, **kwargs):
 
         finally:
             if backup_info:
+
+                # IF is an incremental backup, we save here child backup info id
+                # inside the parent list of children. no matter if the backup
+                # is successful or not. This is needed to be able to retrieve
+                # also failed incremental backups for removal or other operations
+                # like show-backup.
+                parent_backup_info = backup_info.get_parent_backup_info()
+
+                if parent_backup_info:
+                    if parent_backup_info.children_backup_ids:
+                        parent_backup_info.children_backup_ids.append(  # type: ignore
+                            backup_info.backup_id
+                        )
```

Comment:
This is not working because currently the value is stored as text.

Reply:
I added a couple of functions to handle dump and load of backup ids.

Reply:
yeah I realized that this was necessary later after I posted this code snippet, something similar to what happens for tablespaces